imagefun/imagefun.go

```diff
@@ -0,0 +1,255 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"image"
+	"image/color"
+	"image/png"
+	"log"
+	"math/rand"
+	"os"
+	"strings"
+	"time"
+)
+
+// The agenda:
+
+// struct
+// errors
+// panics (Aaaaaaaaa!)
+// arrays & slices
+// maps
+// pointers
+
+const (
+	widthDefault    = 200
+	heightDefault   = 200
+	fileNameDefault = "image.png"
+)
+
+var (
+	colors = map[int]*color.RGBA{
+		// RGBa: RED, GREEN, BLUE, ALPHA ch. (transparency)
+		0:  {R: 100, G: 200, B: 200, A: 0xff},
+		1:  {R: 70, G: 70, B: 21, A: 0xff},
+		2:  {R: 207, G: 70, B: 110, A: 0xff},
+		3:  {R: 78, G: 70, B: 207, A: 0xff},
+		4:  {R: 207, G: 205, B: 70, A: 0xff},
+		5:  {R: 177, G: 37, B: 180, A: 0xff},
+		6:  {R: 225, G: 39, B: 232, A: 0xff},
+		7:  {R: 255, G: 0, B: 0, A: 0xff},
+		8:  {R: 0, G: 255, B: 0, A: 0xff},
+		9:  {R: 0, G: 0, B: 255, A: 0xff},
+		10: {R: 255, G: 128, B: 0, A: 0xff},
+		11: {R: 255, G: 0, B: 127, A: 0xff},
+	}
+	err   error
+	lines []string
+)
+
+type imgOptions struct {
+	width    int
+	height   int
+	colors   map[int]*color.RGBA
+	fileName string
+	content  [][]*color.RGBA
+}
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+func main() {
+
+	var w *int = flag.Int("w", widthDefault, "image width in pixels")
+	var h *int = flag.Int("h", heightDefault, "image height in pixels")
+	var fn *string = flag.String("f", fileNameDefault, "file name")
+	var isFlag *bool = flag.Bool("flag", false, "draw flag image based on user input")
+	flag.Parse()
+
+	var options imgOptions = imgOptions{
```

Comment:
u can just use := oparator to avoid var options. It can be just like ` imgOptions := imgOptions{}`